### PR TITLE
ML-126: Fix problem with starting and shutting down Event and rtc_engine

### PR DESCRIPTION
**Pull request by LVala**

### CODE REVIEW

File: lib/membrane_live_web/channels/event_channel.ex [hunk 1]
```diff
@@ -104,13 +118,24 @@ defmodule MembraneLiveWeb.EventChannel do
     end
   end
 
-  defp join_event_stream(socket) do
+  defp get_event_pid(socket) do
```
Comment (Rados13) [inline lib/membrane_live_web/channels/event_channel.ex:121]: I would rename this function as it doesn't return event pid.

**Review by Rados13 — COMMENTED** [review]
Also CI fails.

File: lib/membrane_live/event.ex [hunk 2]
```diff
@@ -210,26 +216,50 @@ defmodule MembraneLive.Event do
...
 
-      {:stop, :normal, state}
-    else
-      {peer_id, _peer_channel_id} =
-        state.peer_channels
-        |> Enum.find(fn {_peer_id, peer_channel_pid} -> peer_channel_pid == pid end)
+      true ->
+        result =
+          state.peer_channels
+          |> Enum.find(fn {_peer_id, peer_channel_pid} -> peer_channel_pid == pid end)
```
[the remark this answers is not in the document]
Comment (LVala) [reply lib/membrane_live/event.ex:245]: Fixed.

File: lib/membrane_live/event.ex [hunk 3]
```diff
@@ -210,26 +216,50 @@ defmodule MembraneLive.Event do
...
-      if state.peer_channels == %{}, do: Engine.terminate(state.rtc_engine)
+            Engine.remove_peer(state.rtc_engine, peer_id)
 
-      {:noreply, state}
+            if is_nil(state.moderator_pid) and state.peer_channels == %{} do
+              Engine.terminate(state.rtc_engine)
+              {:stop, :normal, state}
+            else
+              {:noreply, state}
+            end
```
[the remark this answers is not in the document]
Comment (LVala) [reply lib/membrane_live/event.ex]: Fixed.

File: lib/membrane_live_web/channels/event_channel.ex (hunk 1 above)
Comment (LVala) [reply lib/membrane_live_web/channels/event_channel.ex]: Fixed.

**Comment by LVala** [comment]
CI fails are connected to changes on [compositor branch](https://github.com/membraneframework-labs/membrane_live/tree/LIVE-92-adding-compositor) and will be taken care of in [this PR](https://github.com/membraneframework-labs/membrane_live/pull/82) to which I am merging with.

**Review by Rados13 — APPROVED** [review]

**Pull request merged**